traefik/templates/service-metrics.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if (and (.Values.metrics.prometheus.service).enabled (not .Values.hub.enabled)) -}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: traefik-metrics
+  namespace: {{ template "traefik.namespace" . }}
+  {{- template "traefik.metrics-service-metadata" . }}
+  annotations:
+  {{- with .Values.metrics.prometheus.service.annotations }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  type: ClusterIP
+  selector:
+    {{- include "traefik.metricslabelselector" . | nindent 4 }}
```

Comment:
With a selector on `app.kubernetes.io/component: metrics`, how this service will select a Traefik Pod ?

Reply:
Oh of course - i fixed it and use the default selector labels.

Reply:
The previous test failed, so i removed the expected value (comp: metrics) and it went green. Imho that's exactly the test-case to requested, isn't it?